chia/full_node/mempool_manager.py

```diff
@@ -447,10 +462,10 @@ async def validate_spend_bundle(
                     False,
                     self.peak.timestamp,
                 )
-
-            assert removal_record is not None
+                removal_record_dict[name] = removal_record
+            else:
```

Comment:
Just curious what this new code path does. we didn't have it in the old code and it doesn't seem related to grouping the query

Reply:
this for-loop used to query one spend at a time (at the top of the loop). This was split into:

1. query all spends in one query
2. build a dict of the responses
3. this for-loop

So, some code moved around a bit.

Previously we would have the current `CoinRecord` here, named `removal_record`, because that's what we looked up (either from the DB or from `additions_dict`. We would then populate the `removal_record_dict` (line 453 in the previous version).

In the new version, we've already built the dictionary (in the first for-loop, line 443 in the new version), so now we need to go the other way, and set `removal_record` in order for the last line to update `removal_amount`.